src/ReverseProxy/Middleware/LoadBalancingMiddleware.cs

```diff
@@ -31,23 +31,22 @@ internal class LoadBalancingMiddleware
 
         public Task Invoke(HttpContext context)
         { 
-            var cluster = context.GetRequiredCluster();
-            var destinationsFeature = context.GetRequiredDestinationFeature();
-            var destinations = destinationsFeature.Destinations;
+            var reverseProxy = context.GetReverseProxyFeature();
```

Comment:
```suggestion
            var proxyFeature = context.GetReverseProxyFeature();
```
or reverseProxyFeature.